File: modules/app/src/main/java/com/prevengos/plug/app/usecase/RegistrarResultadoUseCase.java
package com.prevengos.plug.app.usecase;

import com.prevengos.plug.app.dto.RegistrarResultadoCommand;
import com.prevengos.plug.domain.events.DomainEventPublisher;
import com.prevengos.plug.domain.events.ResultadoRegistradoEvent;
import com.prevengos.plug.domain.model.ResultadoAnalitico;
import com.prevengos.plug.domain.ports.ResultadosPort;

import java.util.Objects;

public class RegistrarResultadoUseCase {
    private final ResultadosPort resultadosPort;
    private final DomainEventPublisher eventPublisher;

    public RegistrarResultadoUseCase(ResultadosPort resultadosPort, DomainEventPublisher eventPublisher) {
        this.resultadosPort = Objects.requireNonNull(resultadosPort, "resultadosPort");
        this.eventPublisher = Objects.requireNonNull(eventPublisher, "eventPublisher");
    }

    public ResultadoAnalitico execute(RegistrarResultadoCommand command) {
        ResultadoAnalitico resultado = resultadosPort.registrar(command.toResultado());
        eventPublisher.publish(new ResultadoRegistradoEvent(command.fuente(), resultado));
        return resultado;
    }
}

File: modules/app/src/main/kotlin/com/prevengos/plug/app/usecase/RegistrarResultadoUseCase.kt
package com.prevengos.plug.app.usecase

import com.prevengos.plug.app.dto.RegistrarResultadoCommand
import com.prevengos.plug.domain.events.DomainEventPublisher
import com.prevengos.plug.domain.events.ResultadoRegistradoEvent
import com.prevengos.plug.domain.model.ResultadoAnalitico
import com.prevengos.plug.domain.ports.ResultadosPort

class RegistrarResultadoUseCase(
    private val resultadosPort: ResultadosPort,
    private val eventPublisher: DomainEventPublisher
) {
    fun execute(command: RegistrarResultadoCommand): ResultadoAnalitico {
        val resultado = resultadosPort.registrar(command.toResultado())
        eventPublisher.publish(
            ResultadoRegistradoEvent(
                source = command.fuente,
                resultado = resultado
            )
        )
        return resultado
    }
}
